src/map.cpp

```diff
@@ -544,37 +544,39 @@ bool Map::isTileClear(uint16_t x, uint16_t y, uint8_t z, bool blockFloor /*= fal
 }
 
 namespace {
-	bool checkSteepLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z)
-	{
-		float dx = x1 - x0;
-		float slope = (dx == 0) ? 1 : (y1 - y0) / dx;
-		float yi = y0 + slope;
-
-		for (uint16_t x = x0 + 1; x < x1; ++x) {
-			if (!g_game.map.isTileClear(std::floor(yi), x, z)) {
-				return false;
-			}
-			yi += slope;
-		}
 
-		return true;
+bool checkSteepLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z)
+{
+	float dx = x1 - x0;
+	float slope = (dx == 0) ? 1 : (y1 - y0) / dx;
+	float yi = y0 + slope;
+
+	for (uint16_t x = x0 + 1; x < x1; ++x) {
+		if (!g_game.map.isTileClear(std::floor(yi), x, z)) {
```

Comment:
This is still probably incorrect. You are using y in place of x. Is this the correct code?

Reply:
Judging by amount of tests I performed throwing in every possible direction without any bugs, it is.